### ISSUE TITLE
Broken rendering with Iris when shaders are enabled

### ISSUE THREAD

**Issue by Nik-mmzd** [Hunternif/AntiqueAtlas#416]
**Describe the bug**
Atlas rendering bug (see screenshots)

**To Reproduce**
Compile for 1.18.1 from 37038a399ecac1d58bcc7164ef3d309e8636a2cb
Install Sodium and Iris
Enable shaders (e. g. BSL)

**Expected behavior**
![image](https://user-images.githubusercontent.com/19509291/146686144-2e08a01a-b9df-49ed-9045-0dc490d30b21.png)
![image](https://user-images.githubusercontent.com/19509291/146686152-1ea97bb4-5c6f-42c8-bd96-94d42b015a69.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/19509291/146686170-ad7383f5-f9a6-40ad-a466-15a801a01327.png)
![image](https://user-images.githubusercontent.com/19509291/146686162-6f267773-f976-46e9-bae8-e2ad68902232.png)

**Fabric version information:**
 - Minecraft: 1.18.1
 - Fabric Loader: 0.12.12
 - Fabric API: 0.44.0+1.18
 - AntiqueAtlas: built from commit 37038a399ecac1d58bcc7164ef3d309e8636a2cb

**Additional context**
When Iris is installed, but no shaders are enabled, all renders normally
Sodium: 0.4.0-alpha5 (latest)
Iris: 1.1.3 (latest)

**Comment by tyra314** [comment]
Does this also happen with the release version?

**Comment by pikasalt** [comment]
Yes, I can confirm that this happens using the latest 1.18.2 release versions of  iris, sodium, and antique atlas

**Comment by tyra314** [comment]
According to #448 it also happens on forge

**Comment by ShiroYatagami** [comment]
Still a problem on iris 1.4.3, sodium 0.4.1 build 15 & indium 1.0.7 when using shaders but not when disabling shaders.

**Comment by tyra314** [comment]
Fixed in 99330bdf842502cbeac3e5b7d8ec18233cd7a970 thanks to @LangYueMc

**Comment by tyra314** [comment]
Turns out, those patches don't resolve the issue properly.

**Comment by Orgamorsh** [comment]
I made a resource pack that kinda band-aid fixes this. The issue is with pixels with mixed transparency values overwriting other ones. This resource pack makes every pixel have 255 opacity. I think it works pretty well. I tested it with oculus on complementary.

[Antique Atlas Shader Patch.zip](https://github.com/user-attachments/files/15754239/Antique.Atlas.Shader.Patch.zip)

![Screenshot 2024-06-09 142902](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/assets/58994640/88528569-86a8-43ee-9e0f-6536909241bf)
